### ISSUE TITLE
Google Analytics

### ISSUE THREAD

**Issue by fattaccioli** [jekyll/minima#816]
Hi there, 

I followed the instructions to use google analytics, and GA does not see any connection to the website, any view, etc.

Could you tell me if the publisehd process is not valid anymore ?

Thanks

**Comment by iTiamo** [comment]
I checked the Analytics dashboard for manual installation instructions for Google Analytics. It provided me with a snippet to include in my site. I compared it to `_includes/google-analytics.html` included in this repo and it did not match. I replaced this file by creating a file called `google-analytics` in my local directory `_includes` with the snippet that was provided to me, which is at time of writing:

```
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-BD3KKE8T6Y"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-BD3KKE8T6Y');
</script>
```

This caused data to start flowing into my Google Analytics instance. If you need to troubleshoot, further check out this documentation: https://support.google.com/tagassistant/answer/10039345?sjid=15349293314530816036-EU.

**Comment by ashmaroli** [comment]
@iTiamo Will you be able to submit a pull request to update the analytics code provided by Minima?

**Comment by fattaccioli** [comment]
site.google_analytics : analytics ID should be added to config.yml right ?

**Comment by iTiamo** [comment]
Yes, this is already detailed in the README. 

https://github.com/jekyll/minima/tree/2.5-stable?tab=readme-ov-file#enabling-google-analytics

**Comment by fattaccioli** [comment]
Yes indeed sorry. Still not working. Maybe a "production"environment variable issue on github. Analytics is only working in production mode according to the README
This is a bit of an annoying struggle.

**Comment by iTiamo** [comment]
Yes, I had to include the environment variable in my GitHub pages deployment. Additionally the Google Analytics tag should be in the root of the config file. Note that the site object is populated from there. So don't create a site object in the config file yourself.

**Comment by fattaccioli** [comment]
How do you include it ? 
From what I got, I thought github was automatically putting the environment variable to production by default.

**Comment by iTiamo** [comment]
If you are using GH pages, you can find the environment variables in your repository's settings, environments, github-pages. I hope that helps.

**Comment by Polovinkin** [comment]
to be honest, there is a problem that readme description of how to add G analytics to config is quite vague, I've made a PR to fix that https://github.com/jekyll/minima/pull/853 more than a week ago, nothing happened for now

when I did it like I've described, stats started flowing to Google Analytics just fine fyi
